drivers/spi/Kconfig.mcux_flexcomm

```diff
@@ -1,9 +1,18 @@
 # Copyright (c) 2016, Freescale Semiconductor, Inc.
-# Copyright (c) 2017,2019, NXP
+# Copyright (c) 2017-2020, NXP
 # SPDX-License-Identifier: Apache-2.0
 
 config SPI_MCUX_FLEXCOMM
 	bool "MCUX FLEXCOMM SPI driver"
 	depends on HAS_MCUX_FLEXCOMM
 	help
 	  Enable support for mcux flexcomm spi driver.
+
+if SPI_MCUX_FLEXCOMM
+config SPI_MCUX_FLEXCOMM_DMA
+	bool "MCUX FLEXCOMM SPI DMA Support"
+	select DMA
```

Comment:
This should probably be `depends on` instead of `select`

Reply:
This saves the user one step of having to add CONFIG_DMA=y

Reply:
`select` can cause a lot of problems, see https://docs.zephyrproject.org/latest/guides/kconfig/tips.html#select-pitfalls

Reply:
I see the pitfalls listed. However in this case if I switch to `depends on` and if a user forgets to add CONFIG_DMA, then the transfers will be interrupt based without the user's knowledge.